app/controllers/media_objects_controller.rb

```diff
@@ -54,6 +55,26 @@ def confirm_remove
     raise CanCan::AccessDenied unless Array(params[:id]).any? { |id| current_ability.can? :destroy, MediaObject.find(id) }
   end
 
+  def intercom_collections
+    respond_to do |format|
+      format.json do
+        render json: Avalon::Intercom.user_collections(user_key).to_json
+      end
+    end
+  end
+
+  def intercom_push
+    result = Avalon::Intercom.push_media_object(@media_object, params[:collection_id], params[:include_structure] == 'true')
+    if result['id'].present?
+      target_url = URI.join(Avalon::Intercom.avalons['default']['url'], 'media_objects/', result['id'])
```

Comment:
Could `Intercom.push_media_object` return this value so it doesn't have to be computed here?

Reply:
It could.

Reply:
I think this might actually be necessary since the line here assumes the default avalon while an instance could have multiple configured.